dotnet/src/SemanticKernel.Abstractions/SemanticFunctions/PromptConfig.cs

```diff
@@ -0,0 +1,278 @@
+
+    #region internal
+
+    /// <summary>
+    /// Prompt template
+    /// </summary>
+    internal IPromptTemplate? PromptTemplate { get; private set; }
+
+    /// <summary>
+    /// Return the list of parameters used by the function as a list of ParameterView instances.
+    /// </summary>
+    /// <returns>List of parameters</returns>
+    internal IList<ParameterView> GetParameters()
+    {
+        List<ParameterView> parameters = new();
+        foreach (var p in this.InputParameters)
+        {
+            parameters.Add(new ParameterView(p.Name, p.Description, p.DefaultValue));
+        }
+        return parameters;
+    }
+
+    /// <summary>
+    /// Render the template using the information in the context
+    /// </summary>
+    /// <param name="engine">Prompt template engine</param>
+    /// <param name="executionContext">Kernel execution context helpers</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>Prompt rendered to string</returns>
+    internal Task<string> RenderAsync(IPromptTemplateEngine engine, SKContext executionContext, CancellationToken cancellationToken = default)
+    {
+        if (this.PromptTemplate is not null)
+        {
+            return this.PromptTemplate.RenderAsync(executionContext, cancellationToken);
+        }
+
+        return engine.RenderAsync(this.GetTemplate(), executionContext, cancellationToken);
+    }
+
+    /// <summary>
+    /// Get the prompt template.
+    /// </summary>
+    /// <returns>Prompt template</returns>
+    internal string GetTemplate()
+    {
+        if (this.Template is null)
+        {
+            throw new KernelException(
+                KernelException.ErrorCodes.InvalidTemplate,
+                "A template cannot be null.");
+        }
+
+        // TODO Add support for loading from the template path
+
+        return this.Template;
+    }
+
+    /// <summary>
+    /// Get the default request settings to be used when the associated prompt is executed.
+    /// </summary>
+    /// <returns>JSON string containing the default request settings</returns>
+    internal string? GetDefaultRequestSettingsAsJson()
+    {
+        if (this.RequestSettings is not null && this.RequestSettings.Count > 0)
+        {
+            // Initial implementation only supports a single instance of request settings
+            // and the first item in the list will be the default settings.
+            // Later this pattern will be extended to allow the request settings to be selected
+            // when the semantic function is invoked.
+            var defaultSettings = this.RequestSettings[0];
+            return defaultSettings.Properties.ToJson();
+        }
+
+        return null;
+    }
+
+    /// <summary>
+    /// Convert SemanticFunctionConfig to the new abstraction PromptConfig
```

Comment:
nit: If you use `see` attribute for class names, it will be easier to maintain XML documentation in the future, when you rename some class or delete it for example. Same applies to other places in the code where class names are used. Can be done in follow-up PRs.
```suggestion
    /// Convert <see cref="SemanticFunctionConfig"/> to the new abstraction <see cref="PromptConfig"/>
```